apps/build-onchain-apps/src/contract/ContractSpecification.ts

```diff
@@ -0,0 +1,26 @@
+import { Abi, type Chain } from 'viem';
+import { baseGoerli } from 'viem/chains';
+import BuyMeACoffeeABI from './BuyMeACoffee';
+
+export type ContractInstance = {
```

Comment:
if we are exporting `types`, should at this point have just have a `types.ts` file in this folder that we use to organize different kind of Contract types we need?

Reply:
I usually prefer to have the types colocated with any implementation code, but open to your approach as well. Do you have a strong preference?

Reply:
Approved and merged the PR as this is not a blocker.

My thinking is about where and how things can be re-use. I don't want to jump on conclusion, so happy to just revisit this as we have more code to read :-)

LFG